update sail to 1.13, new dockerfiles to work with 8.1

jxjj opened the pull request:
Updates Laravel Sail from 1.12 to 1.13 <s>and downgrades sail's php from 8.1 to 7.4. (It looks like we're currently running with 7.3 on production).</s>

EDIT: This PR no longer downgrade's sail's php.

cmcfadden commented:
Can we just move to PHP 8.1? Updating prod is easy peasy, and we need to be on 8.1 for L9.

jxjj commented:
Okay:
- Now on php 8.1
- on deploy, composer ignores platform requirements for packages that don't like php@8.1. We can remove this once we've updated packages.
- scoping some webpack dev stuff (e.g. cert keys for webpack dev server) to non-production environments
- removed node-sass since we already have the preferred dart sass

@cmcfadden – Thanks for your help with this. Would you mind looking this over again to see if we want to make any changes before merging?

cmcfadden commented:
looks good! one more step towards modernity!
